### PR TITLE
tracing: Add SkyWalking tracer

### PR DESCRIPTION
**Pull request by dio**
This is a WIP of adding SkyWalking

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by dio** [comment]
cc. @ yskopets

File: CODEOWNERS [hunk 1]
```diff
@@ -48,6 +48,8 @@ extensions/filters/common/original_src @snowp @klarose
 /*/extensions/tracers/datadog @cgilmour @palazzem @mattklein123
 # tracers.xray extension
 /*/extensions/tracers/xray @marcomagdy @lavignes @mattklein123
+# tracers.skywalking extension
+/*/extensions/tracers/skywalking @dio @lizan
```
Comment (dio) [inline CODEOWNERS:52]: We can add more names here.

**Comment by dio** [comment]
So mostly how we model the SegmentObject from spans is still probably wrong. And also need to strictly do the decision on tracing based on the propagated context.

**Comment by dio** [comment]
And yes, missing tests! Haha.

**Comment by yskopets** [comment]
@dio does SkyWalking project has a demo setup we could play with?

File: source/extensions/tracers/skywalking/collector/tracing.proto [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
```
Comment (yskopets) [inline source/extensions/tracers/skywalking/collector/tracing.proto:1]: I think, we will be asked to import this `*.proto` from an external dependency.

similarly to `OpenCensus`.

Comment (dio) [reply source/extensions/tracers/skywalking/collector/tracing.proto:1]: I had that before. That should be easy to fix.

**Comment by dio** [comment]
> @dio does SkyWalking project has a demo setup we could play with?

I can set it up for you.

**Pull request closed without merging**